### ISSUE TITLE
[import/extensions] Doesn't error on . or ..

### ISSUE THREAD

**Issue by benasher44** [import-js/eslint-plugin-import#2777]
I have this rule configured like so:
```javascript
'import/extensions': ['error', 'ignorePackages'],
```

With this configuration, the following is flagged as an error:
```javascript
import { Something } from './LocalFolderWithIndexfile'
```

However, the following is not flagged as an error, but I believe it should be:
```javascript
import { Something } from '.'
```

The only difference between the two is where in the file tree the importing file sits. As a workaround, one can supplement this rule with the general `no-restricted-imports` rule like this:
```javascript
'no-restricted-imports': ['error',
  {
    paths: [
      {
        name: '..',
        message: 'Import from ../index.js instead.',
      },
      {
        name: '.',
        message: 'Import from ./index.js instead.',
      },
    ],
  },
],
```

Apologies if this has already been filed. I found it really hard to search for

**Comment by ljharb** [comment]
`.` should not be, because that's not necessarily pointing to a specific file, it's potentially pointing to a package.json. `.` won't always be `index.js`, and the whole point of that feature is to *not* specify extensions and defer to node's algorithm for it.

**Comment by ljharb** [comment]
Although I do see the discontinuity - when `.` points to an index file and there's no package.json, i think you're right, it should warn.

**Comment by benasher44** [comment]
Yep it's exactly that case: when it's meant to point to an index.json w/o a package.json
